stacks_queues/queue.py

```diff
@@ -23,40 +24,103 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+
+        if (self._front == 0 and self.rear == self.buffer_size - 1) or self.rear == (self._front - 1):
+            raise QueueFullException
+        # queue is empty
+        elif (self._front == -1):
+            self.store_size += 1
+            self._front = 0
+            self.rear = 0
+            self.store[self.rear] = element
+        # wrap rear
+        elif (self.rear == self.buffer_size - 1):
+            self.store_size += 1
+            self.rear = 0
+            self.store[self.rear] = element
+        # rear is not at end of list
+        else:
+            self.store_size += 1
+            self.rear = self.rear + 1
+            self.store[self.rear] = element
+        
+        return None   
 
     def dequeue(self):
         """ Removes an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
             returns None
+
+            Move front pointer and not rear pointer
+            Only change rear when queue becomes empty
         """
-        pass
 
+        # Walkthrough w/Terah:
+        # find index of front
+        # store value in temp value to return
+        # self.store at index of front is going to be None
+        # reassign where front is pointing
+
+        # if self.store_size == 0
+        if self._front == -1:
+            raise QueueEmptyException
+        else:
+            temp = self.store[self._front]
+            self.store[self._front] = None
+            self._front = (self._front + 1) % self.buffer_size
+            self.store_size -=1
+            return temp
+        
+    
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        
+        if (self._front == -1):
+            return None
+
+        return self.store[0]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return len(self.store)
```

Comment:
You're returning the buffer size instead of the number of elements. 

```suggestion
        return self.size
```